src/vm.c

```diff
@@ -774,11 +787,11 @@ v7_val_t v7_create_string(struct v7 *v7, const char *p, size_t len, int own) {
     s[-1] = len;
     tag = V7_TAG_STRING_I;
   } else if (own) {
-    embed_string(m, m->len, p, len);
+    if(embed_string(m, m->len, p, len) < 0) throw_exception(v7, "Error", "Sintax error");
```

Comment:
the first string parameter of `throw_exception` is the error class, and we have "SyntaxError" among valid error types:

`throw_exception(v7, "SyntaxError", "Invalid string")` or something like that